packages/waku/src/lib/vite-rsc/handler.ts

```diff
@@ -121,31 +121,20 @@ const toProcessBuild =
       renderHtml: renderUtils.renderHtml,
       rscPath2pathname: (rscPath) =>
         joinPath(config.rscBase, encodeRscPath(rscPath)),
+      htmlPath2pathname: (htmlPath) =>
+        htmlPath === '/404' ? '404.html' : htmlPath + '/index.html',
       saveBuildMetadata: async (key, value) => {
         buildMetadata.set(key, value);
       },
       withRequest: (req, fn) => INTERNAL_runWithContext(req, fn),
-      generateFile: async (pathname: string, body: ReadableStream | string) => {
-        const filePath = joinPath(
-          DIST_PUBLIC,
-          extname(pathname)
-            ? pathname
-            : pathname === '/404'
-              ? '404.html' // HACK special treatment for 404, better way?
-              : pathname + '/index.html',
-        );
-        await emitFile(filePath, body);
+      generateFile: async (pathname, body) => {
```

Comment:
Let's use `filename` or `fileName` instead of `pathname` which means `URL`'s pathname throughout our codebase.